octoprint_mrbeam/static/js/convert.js

```diff
@@ -308,6 +305,9 @@ $(function () {
                 new_material.device_model = MRBEAM_MODEL;
                 new_material.custom = true;
                 new_material.v = BEAMOS_VERSION;
+                // Remove compatibility keys
+                delete new_material.compatible;
+                delete new_material.customBeforeElementContent;
```

Comment:
Why? I could understand if you would set the keys to be `new_material.compatible = true`. If you don't need the keys, then why have them in the 1st place?

I guess `some_material.compatible = ko.computed(function() { return MRBEAM_LASER_HEAD_MODEL == some_material.laser_head }` ?

Reply:
No, this is a bug I found where upon saving custom materials with the compatibility keys available then these will get saved within the yaml file. So here I'm just preventing that from happening because it would lead to a bug whenever you switch laserheads again and go back to older laserhead.

Reply:
You should probably sanitise that when the custom material gets saved in the back end

Reply:
I would leave it in the frontend as the whole concept of compatibility exist there

Reply:
Please make sure that the owner of the data (in this case the back-end) makes a sanity check on the data it is given. In this case I would filter away any key we do not expect. We could possibly be doing a more extensive check on the types etc but that is not quite required right now

Reply:
in this case it could be
```python
accepted keys = ['name', 'key', ...]
result_sanitised = {}
for mat_key, mat_info in json_from_api_call.items():
  result_sanitised[mat_key] = {k: v for k,v in mat_info.items() if k in accepted_keys}
```

Reply:
N.B. This would also prevent your compatibility bug when changing laserheads.

Reply:
Done